feat: update contribution workflow based on guideline generated as comment (#251)
rustfoundation/safety-critical-rust-coding-guidelines: CONTRIBUTING.md

You would then pull `fls_4rhjpdu4zfqj` to place in the FLS ID field.

### 2) Guideline Generated as a Comment

A GitHub Action will fire, adding a comment to your newly created issue with
the contents of the coding guideline prepared written out correctly

> in reStructured Text.

Reviewer: We should mention that this comment gets updated with every edit. It's a really useful feature you guys pulled off which makes contributing so much easier.

Reviewer: (PS: I would suggest an edit myself, but good god I'm so tired right now)

### 3) Create a Draft with a Member

A member of the Coding Guidelines Subcommittee should get you a first review
with some feedback within 14 days of submission. You'll work with one or more
members to flesh out the concept and ensure the guideline is well prepared.

> ### 4) Contributor Creates the PR

Reviewer: ```suggestion
### 4) Create the PR
```

This fits the voicing of the rest of the workflow better

Once an issue has been well-developed enough, a subcommittee member will mark
the issue with the label `sign-off: create pr` to denote the contributor
should create a PR based on the contents of the latest comment containing
the generated guideline from

> [2) Guideline Generated as a Comment](#2-guideline-generated-as-a-comment).

Reviewer: ```suggestion
the issue with the label `sign-off: create pr` to denote that you
should create a Pull Request.

The contents of the PR should be based on the latest bot comment containing
the generated guideline from
[2) Guideline Generated as a Comment](#2-guideline-generated-as-a-comment).
```

The voicing is more personal elsewhere in the doc, so I'm mainly adjusting for that. The original sentence was also a bit long (it was a composite sentence), so I split it to make it easier to read.

Make sure to include the following in the body of your PR so that the
corresponding issue will be closed upon the PR being closed. Note that
`#xyz` corresponds to the number of the issue opened by
[1) Submit coding guideline issue](#1-submit-coding-guideline-issue).

Follow the guidance given in the comment with the guideline on which

> chapter to place the coding guideline.

Reviewer: I'm not sure what we mean here. What do we mean here? :)

Author: Tried to fix the odd formatting here: https://github.com/rustfoundation/safety-critical-rust-coding-guidelines/pull/251/changes/843e5fea0e745657dd466742ee28a1cf8edc57ed

`closes #xyz`

### 5) Iterate on Feedback

#### 5.1) Member Begins Review

The generated Pull Request may attract additional feedback or simply be an easier place to suggest targeted edits.

As the contributor of the coding guideline and opener of the issue, you'll respond to comments, discuss, all the normal things on the pull request.

#### 5.2) Update the PR Based on Feedback

If you agree with the suggested changes, you've got two options:

> - iterate directly on the Pull Request, if you're comfortable with reStructured Text to do so

Reviewer: ```suggestion
- Iterate directly on the Pull Request, if you're comfortable with reStructured Text to do so
```

> - if you'd rather make revisions in Markdown, you can return to the issue

Reviewer: ```suggestion
- If you'd rather make revisions in Markdown, you can return to the issue
```

  from [1) Submit coding guideline issue](#1-submit-coding-guideline-issue)
  to regenerate the reStructured Text guideline form by following steps
  outlined in
  [6) Contributor Applies Feedback on Issue](#6-contributor-applies-feedback-on-issue)
  and

>   [7) Contributor Applies Regenerated Guideline to PR](#7-contributor-applies-regenerated-guideline-to-pr)

Author: So here's my thought:

* If we think it's worthwhile to have it shown in the flowchart as explicit steps on how to regenerate the reStructuredText via the issue, then we should keep steps 6 and 7 (or rename them to 5.2.2.1 and 5.2.2.2, but that feels bad 🙃)
* If we don't -- I can probably rewrite the text under the 2nd bullet point under 5.2 to pull the details from steps 6, 7 directly into there.

What do you think, @felix91gr?

<img width="2136" height="1580" alt="image" src="https://github.com/user-attachments/assets/08baee20-0bf5-4a67-a433-03d021b2f94f" />

> [!NOTE]
> Here's a list of recommended prerequisites that shall be fulfilled before turning an issue into a PR:
>
> * The new rule isn't already covered by another rule or, in case there is already another rule, the rules need to be linked and categorized
> * All sections contain some content
> * Content written may be *incomplete*, but must not be *incorrect*
> * `🧪 Code Example Test Results` section shows all example code compiles

As soon as these prerequisites are fulfilled, the draft shall be marked as PR-ready by a subcommittee member, by labeling the issue with `sign-off: create pr`. This denotes that you should create a Pull Request with your Guideline. Further discussion about the amount and correctness of its content shall then be done on the Pull Request itself.

The contents of the PR should be based on the bot comment containing the generated RST form of your guideline, as seen in [Step 2](#2-guideline-generated-as-a-comment). The comment has the exact file content you'll need.

Reference the generated comment from [Step 2](#2-guideline-generated-as-a-comment),
in particular headings `📁 Target Location` and `🗂️ Update Chapter Index` for necessary

> steps to ensure that your guideline appears when rendering the guidelines.

Reviewer: ```suggestion
In order to ensure your guideline appears when rendering the document, reference the generated comment from [Step 2](#2-guideline-generated-as-a-comment). All the steps necessary should appear below the headings `📁 Target Location` and `🗂️ Update Chapter Index`.
```

Cutting the composite sentence in two, to make it easier to digest :)

You would then pull `fls_4rhjpdu4zfqj` to place in the FLS ID field.

### 2) Guideline Generated as a Comment

A GitHub Action will fire, adding a comment to your newly created issue with
the contents of the coding guideline prepared written out correctly
in reStructuredText.

Note that if you later update the body of the coding guideline issue this will
fire the GitHub Action again and update the original comment with the new
contents converted to reStructuredText.

### 3) Create a Draft with a Member

Within 14 days of your submission, a member of the Coding Guidelines Subcommittee should give you a first review. You'll work with them (and other members) to flesh out the concept and ensure the guideline is well prepared for a Pull Request.

### 4) Create the PR

Once an issue has been well-developed enough, a subcommittee member will mark
the issue with the label `sign-off: create pr` to denote that you

> should create a Pull Request.

Reviewer 2: Let's add a definition of "well-developed" here:

> Here's a list of recommended prerequisites that shall be fulfilled before turning an issue into a PR:
> * The new rule isn't already covered by another rule or, in case there is already another rule, the rules need to be linked and categorized
> * All sections contain some content
> 
> As soon as these are fulfilled, the issue shall be converted into a PR by a subcommittee member by marking the issue with the label `sign-off: create pr` to denote that you should create a Pull Request. Further discussion about the amount and correctness of the content shall then be done on the Pull Request and the Issue gets closed.

You would then pull `fls_4rhjpdu4zfqj` to place in the FLS ID field.

### 2) Guideline Generated as a Comment

A GitHub Action will fire, adding a comment to your newly created issue with
the contents of the coding guideline prepared written out correctly
in reStructuredText.

Note that if you later update the body of the coding guideline issue this will
fire the GitHub Action again and update the original comment with the new
contents converted to reStructuredText.

### 3) Create a Draft with a Member

Within 14 days of your submission, a member of the Coding Guidelines Subcommittee should give you a first review. You'll work with them (and other members) to flesh out the concept and ensure the guideline is well prepared for a Pull Request.

### 4) Create the PR

> [!NOTE]
> Here's a list of recommended prerequisites that shall be fulfilled before turning an issue into a PR:
>

> > * The new rule isn't already covered by another rule or, in case there is already another rule, the rules need to be linked and categorized

Reviewer: ```suggestion
> * The new rule isn't already covered by another rule
>     * OR, in case there is already another rule(s),
>     * The existing rule(s) need(s) to be linked to the new rule,
>     * AND the new rule needs to link to the existing rule(s).
```

Making it easier to read. Since it's a list of requirements, I reckon it's good to have a piece-by-piece list.

What do you think, @PLeVasseur?

> * All sections contain some content
> * Content written may be *incomplete*, but must not be *incorrect*

As soon as these prerequisites are fulfilled, the issue shall be converted into a PR by a subcommittee member by marking the issue with the label `sign-off: create pr` to denote that you should create a Pull Request. Further discussion about the amount and correctness of the content shall then be done on the Pull Request.

> The contents of the PR should be based on the bot comment containing the generated RST form of your guideline, as seen in [Step 2](#2-guideline-generated-as-a-comment) as the exact file content you'll need.

Reviewer: ```suggestion
The contents of the PR should be based on the bot comment containing the generated RST form of your guideline, as seen in [Step 2](#2-guideline-generated-as-a-comment). The comment has the exact file content you'll need.
```
